{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 98,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "from subprocess import call\n",
    "from ast import literal_eval"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [],
   "source": [
    "def isNaN(num):\n",
    "    return num != num"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "lactate_table = pd.read_csv(\"/Users/wassilyminkow/Downloads/HCAR1_orthologs.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "common_name = np.asarray(lactate_table[\"Common name\"])\n",
    "# scientific_name = list(map(lambda x: x.replace(\" \",\"_\"), scientific_name))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "isNaN(common_name[-2])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "nan"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "common_name[-2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [],
   "source": [
    "i=0\n",
    "with open('/Users/wassilyminkow/Downloads/HCAR1_gene_nuccore_refseqrna.fasta', 'r') as reader:\n",
    "    new = []\n",
    "    for line in reader.readlines():\n",
    "        if line[0] == \">\":\n",
    "            line = line[0]+scientific_name[i]+\" \"+line[1:]\n",
    "            i+=1\n",
    "        new.append(line)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [],
   "source": [
    "with open('/Users/wassilyminkow/Downloads/HCAR1_gene_nuccore_refseqrna_mod.fasta', 'w') as writer:\n",
    "    for line in new:\n",
    "        writer.write(line)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [],
   "source": [
    "students = pd.read_excel(\"/Users/wassilyminkow/Downloads/pan-genome/CG19_pangenomics.xlsx\", sheet_name=\"SK_students_CompGen19\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [],
   "source": [
    "strains = pd.read_excel(\"/Users/wassilyminkow/Downloads/pan-genome/CG19_pangenomics.xlsx\", sheet_name=\"list of strains\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>N</th>\n",
       "      <th>Strain name</th>\n",
       "      <th>Assembly accession</th>\n",
       "      <th>Gbrs_paired_asm</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>Bacillus amyloliquefaciens DSM 7</td>\n",
       "      <td>GCA_000196735.1</td>\n",
       "      <td>GCF_000196735.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>Bacillus amyloliquefaciens IT-45</td>\n",
       "      <td>GCA_000242855.2</td>\n",
       "      <td>GCF_000242855.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>Bacillus amyloliquefaciens LL3</td>\n",
       "      <td>GCA_000204275.1</td>\n",
       "      <td>GCF_000204275.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>Bacillus amyloliquefaciens TA208</td>\n",
       "      <td>GCA_000195515.1</td>\n",
       "      <td>GCF_000195515.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5</td>\n",
       "      <td>Bacillus amyloliquefaciens XH7</td>\n",
       "      <td>GCA_000221645.1</td>\n",
       "      <td>GCF_000221645.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>6</td>\n",
       "      <td>Bacillus anthracis str. A0248</td>\n",
       "      <td>GCA_000022865.1</td>\n",
       "      <td>GCF_000022865.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>7</td>\n",
       "      <td>Bacillus anthracis str. Ames</td>\n",
       "      <td>GCA_000007845.1</td>\n",
       "      <td>GCF_000007845.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>8</td>\n",
       "      <td>Bacillus anthracis str. CDC 684</td>\n",
       "      <td>GCA_000021445.1</td>\n",
       "      <td>GCF_000021445.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>9</td>\n",
       "      <td>Bacillus anthracis str. H9401</td>\n",
       "      <td>GCA_000258885.1</td>\n",
       "      <td>GCF_000258885.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>10</td>\n",
       "      <td>Bacillus anthracis str. Sterne</td>\n",
       "      <td>GCA_000008165.1</td>\n",
       "      <td>GCF_000008165.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>11</td>\n",
       "      <td>Bacillus anthracis str. Vollum</td>\n",
       "      <td>GCA_000742895.1</td>\n",
       "      <td>GCF_000742895.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>12</td>\n",
       "      <td>Bacillus cereus 03BB102</td>\n",
       "      <td>GCA_000022505.1</td>\n",
       "      <td>GCF_000022505.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>13</td>\n",
       "      <td>Bacillus cereus 03BB108</td>\n",
       "      <td>GCA_000832865.1</td>\n",
       "      <td>GCF_000832865.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>14</td>\n",
       "      <td>Bacillus cereus 172560W</td>\n",
       "      <td>GCA_000160935.1</td>\n",
       "      <td>GCF_000160935.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>15</td>\n",
       "      <td>Bacillus cereus 95/8201</td>\n",
       "      <td>GCA_000161135.1</td>\n",
       "      <td>GCF_000161135.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>16</td>\n",
       "      <td>Bacillus cereus AH1271</td>\n",
       "      <td>GCA_000161375.1</td>\n",
       "      <td>GCF_000161375.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>17</td>\n",
       "      <td>Bacillus cereus AH1272</td>\n",
       "      <td>GCA_000161395.1</td>\n",
       "      <td>GCF_000161395.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>18</td>\n",
       "      <td>Bacillus cereus AH1273</td>\n",
       "      <td>GCA_000003955.1</td>\n",
       "      <td>GCF_000003955.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>19</td>\n",
       "      <td>Bacillus cereus AH187</td>\n",
       "      <td>GCA_000021225.1</td>\n",
       "      <td>GCF_000021225.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>20</td>\n",
       "      <td>Bacillus cereus AH603</td>\n",
       "      <td>GCA_000161335.1</td>\n",
       "      <td>GCF_000161335.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>21</td>\n",
       "      <td>Bacillus cereus AH621</td>\n",
       "      <td>GCA_000160975.1</td>\n",
       "      <td>GCF_000160975.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>22</td>\n",
       "      <td>Bacillus cereus AH676</td>\n",
       "      <td>GCA_000161355.1</td>\n",
       "      <td>GCF_000161355.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>23</td>\n",
       "      <td>Bacillus cereus AH820</td>\n",
       "      <td>GCA_000021785.1</td>\n",
       "      <td>GCF_000021785.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>24</td>\n",
       "      <td>Bacillus cereus ATCC 10876</td>\n",
       "      <td>GCA_000160895.1</td>\n",
       "      <td>GCF_000160895.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>25</td>\n",
       "      <td>Bacillus cereus ATCC 10987</td>\n",
       "      <td>GCA_000008005.1</td>\n",
       "      <td>GCF_000008005.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>26</td>\n",
       "      <td>Bacillus cereus ATCC 14579</td>\n",
       "      <td>GCA_000007825.1</td>\n",
       "      <td>GCF_000007825.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>27</td>\n",
       "      <td>Bacillus cereus ATCC 4342</td>\n",
       "      <td>GCA_000161015.1</td>\n",
       "      <td>GCF_000161015.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>28</td>\n",
       "      <td>Bacillus cereus B4264</td>\n",
       "      <td>GCA_000021205.1</td>\n",
       "      <td>GCF_000021205.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>29</td>\n",
       "      <td>Bacillus cereus BDRD-Cer4</td>\n",
       "      <td>GCA_000161115.1</td>\n",
       "      <td>GCF_000161115.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>30</td>\n",
       "      <td>Bacillus cereus BDRD-ST196</td>\n",
       "      <td>GCA_000161095.1</td>\n",
       "      <td>GCF_000161095.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65</th>\n",
       "      <td>66</td>\n",
       "      <td>Bacillus subtilis subsp. subtilis str. 168</td>\n",
       "      <td>GCA_000009045.1</td>\n",
       "      <td>GCF_000009045.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66</th>\n",
       "      <td>67</td>\n",
       "      <td>Bacillus subtilis subsp. subtilis str. BSP1</td>\n",
       "      <td>GCA_000321395.1</td>\n",
       "      <td>GCF_000321395.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67</th>\n",
       "      <td>68</td>\n",
       "      <td>Bacillus subtilis subsp. subtilis str. RO-NN-1</td>\n",
       "      <td>GCA_000227485.1</td>\n",
       "      <td>GCF_000227485.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>68</th>\n",
       "      <td>69</td>\n",
       "      <td>Bacillus thuringiensis BMB171</td>\n",
       "      <td>GCA_000092165.1</td>\n",
       "      <td>GCF_000092165.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69</th>\n",
       "      <td>70</td>\n",
       "      <td>Bacillus thuringiensis Bt407</td>\n",
       "      <td>GCA_000161495.1</td>\n",
       "      <td>GCF_000161495.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70</th>\n",
       "      <td>71</td>\n",
       "      <td>Bacillus thuringiensis HD-771</td>\n",
       "      <td>GCA_000292455.1</td>\n",
       "      <td>GCF_000292455.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71</th>\n",
       "      <td>72</td>\n",
       "      <td>Bacillus thuringiensis HD-789</td>\n",
       "      <td>GCA_000292705.1</td>\n",
       "      <td>GCF_000292705.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>72</th>\n",
       "      <td>73</td>\n",
       "      <td>Bacillus thuringiensis IBL 200</td>\n",
       "      <td>GCA_000161715.1</td>\n",
       "      <td>GCF_000161715.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>73</th>\n",
       "      <td>74</td>\n",
       "      <td>Bacillus thuringiensis MC28</td>\n",
       "      <td>GCA_000300475.1</td>\n",
       "      <td>GCF_000300475.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74</th>\n",
       "      <td>75</td>\n",
       "      <td>Bacillus thuringiensis serovar andalousiensis ...</td>\n",
       "      <td>GCA_000161635.1</td>\n",
       "      <td>GCF_000161635.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75</th>\n",
       "      <td>76</td>\n",
       "      <td>Bacillus thuringiensis serovar berliner ATCC 1...</td>\n",
       "      <td>GCA_000161615.1</td>\n",
       "      <td>GCF_000161615.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76</th>\n",
       "      <td>77</td>\n",
       "      <td>Bacillus thuringiensis serovar chinensis CT-43</td>\n",
       "      <td>GCA_000193355.1</td>\n",
       "      <td>GCF_000193355.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>77</th>\n",
       "      <td>78</td>\n",
       "      <td>Bacillus thuringiensis serovar finitimus YBT-020</td>\n",
       "      <td>GCA_000190515.1</td>\n",
       "      <td>GCF_000190515.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78</th>\n",
       "      <td>79</td>\n",
       "      <td>Bacillus thuringiensis serovar huazhongensis B...</td>\n",
       "      <td>GCA_000161675.1</td>\n",
       "      <td>GCF_000161675.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79</th>\n",
       "      <td>80</td>\n",
       "      <td>Bacillus thuringiensis serovar kurstaki str. HD73</td>\n",
       "      <td>GCA_000338755.1</td>\n",
       "      <td>GCF_000338755.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80</th>\n",
       "      <td>81</td>\n",
       "      <td>Bacillus thuringiensis serovar kurstaki str. T...</td>\n",
       "      <td>GCA_000161575.1</td>\n",
       "      <td>GCF_000161575.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>81</th>\n",
       "      <td>82</td>\n",
       "      <td>Bacillus thuringiensis serovar monterrey BGSC ...</td>\n",
       "      <td>GCA_000161595.1</td>\n",
       "      <td>GCF_000161595.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82</th>\n",
       "      <td>83</td>\n",
       "      <td>Bacillus thuringiensis serovar pakistani str. ...</td>\n",
       "      <td>GCA_000161555.1</td>\n",
       "      <td>GCF_000161555.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83</th>\n",
       "      <td>84</td>\n",
       "      <td>Bacillus thuringiensis serovar pondicheriensis...</td>\n",
       "      <td>GCA_000161655.1</td>\n",
       "      <td>GCF_000161655.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>84</th>\n",
       "      <td>85</td>\n",
       "      <td>Bacillus thuringiensis serovar pulsiensis BGSC...</td>\n",
       "      <td>GCA_000161695.1</td>\n",
       "      <td>GCF_000161695.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>85</th>\n",
       "      <td>86</td>\n",
       "      <td>Bacillus thuringiensis serovar thuringiensis s...</td>\n",
       "      <td>GCA_000341665.1</td>\n",
       "      <td>GCF_000341665.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86</th>\n",
       "      <td>87</td>\n",
       "      <td>Bacillus thuringiensis serovar thuringiensis s...</td>\n",
       "      <td>GCA_000161515.1</td>\n",
       "      <td>GCF_000161515.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>87</th>\n",
       "      <td>88</td>\n",
       "      <td>Bacillus thuringiensis serovar tochigiensis BG...</td>\n",
       "      <td>GCA_000161475.1</td>\n",
       "      <td>GCF_000161475.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>88</th>\n",
       "      <td>89</td>\n",
       "      <td>Bacillus thuringiensis str. Al Hakam</td>\n",
       "      <td>GCA_000015065.1</td>\n",
       "      <td>GCF_000015065.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89</th>\n",
       "      <td>90</td>\n",
       "      <td>Bacillus velezensis AS43.3</td>\n",
       "      <td>GCA_000319475.1</td>\n",
       "      <td>GCF_000319475.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>90</th>\n",
       "      <td>91</td>\n",
       "      <td>Bacillus velezensis CAU B946</td>\n",
       "      <td>GCA_000283695.1</td>\n",
       "      <td>GCF_000283695.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91</th>\n",
       "      <td>92</td>\n",
       "      <td>Bacillus velezensis FZB42</td>\n",
       "      <td>GCA_000015785.1</td>\n",
       "      <td>GCF_000015785.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>92</th>\n",
       "      <td>93</td>\n",
       "      <td>Bacillus velezensis UCMB5036</td>\n",
       "      <td>GCA_000341875.1</td>\n",
       "      <td>GCF_000341875.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>93</th>\n",
       "      <td>94</td>\n",
       "      <td>Bacillus velezensis YAU B9601-Y2</td>\n",
       "      <td>GCA_000284395.1</td>\n",
       "      <td>GCF_000284395.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>94</th>\n",
       "      <td>95</td>\n",
       "      <td>Bacillus xiamenensis</td>\n",
       "      <td>GCA_001857925.1</td>\n",
       "      <td>GCF_001857925.1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>95 rows Ã— 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     N                                        Strain name Assembly accession  \\\n",
       "0    1                   Bacillus amyloliquefaciens DSM 7    GCA_000196735.1   \n",
       "1    2                   Bacillus amyloliquefaciens IT-45    GCA_000242855.2   \n",
       "2    3                     Bacillus amyloliquefaciens LL3    GCA_000204275.1   \n",
       "3    4                   Bacillus amyloliquefaciens TA208    GCA_000195515.1   \n",
       "4    5                     Bacillus amyloliquefaciens XH7    GCA_000221645.1   \n",
       "5    6                      Bacillus anthracis str. A0248    GCA_000022865.1   \n",
       "6    7                       Bacillus anthracis str. Ames    GCA_000007845.1   \n",
       "7    8                    Bacillus anthracis str. CDC 684    GCA_000021445.1   \n",
       "8    9                      Bacillus anthracis str. H9401    GCA_000258885.1   \n",
       "9   10                     Bacillus anthracis str. Sterne    GCA_000008165.1   \n",
       "10  11                     Bacillus anthracis str. Vollum    GCA_000742895.1   \n",
       "11  12                            Bacillus cereus 03BB102    GCA_000022505.1   \n",
       "12  13                            Bacillus cereus 03BB108    GCA_000832865.1   \n",
       "13  14                            Bacillus cereus 172560W    GCA_000160935.1   \n",
       "14  15                            Bacillus cereus 95/8201    GCA_000161135.1   \n",
       "15  16                             Bacillus cereus AH1271    GCA_000161375.1   \n",
       "16  17                             Bacillus cereus AH1272    GCA_000161395.1   \n",
       "17  18                             Bacillus cereus AH1273    GCA_000003955.1   \n",
       "18  19                              Bacillus cereus AH187    GCA_000021225.1   \n",
       "19  20                              Bacillus cereus AH603    GCA_000161335.1   \n",
       "20  21                              Bacillus cereus AH621    GCA_000160975.1   \n",
       "21  22                              Bacillus cereus AH676    GCA_000161355.1   \n",
       "22  23                              Bacillus cereus AH820    GCA_000021785.1   \n",
       "23  24                         Bacillus cereus ATCC 10876    GCA_000160895.1   \n",
       "24  25                         Bacillus cereus ATCC 10987    GCA_000008005.1   \n",
       "25  26                         Bacillus cereus ATCC 14579    GCA_000007825.1   \n",
       "26  27                          Bacillus cereus ATCC 4342    GCA_000161015.1   \n",
       "27  28                              Bacillus cereus B4264    GCA_000021205.1   \n",
       "28  29                          Bacillus cereus BDRD-Cer4    GCA_000161115.1   \n",
       "29  30                         Bacillus cereus BDRD-ST196    GCA_000161095.1   \n",
       "..  ..                                                ...                ...   \n",
       "65  66         Bacillus subtilis subsp. subtilis str. 168    GCA_000009045.1   \n",
       "66  67        Bacillus subtilis subsp. subtilis str. BSP1    GCA_000321395.1   \n",
       "67  68     Bacillus subtilis subsp. subtilis str. RO-NN-1    GCA_000227485.1   \n",
       "68  69                      Bacillus thuringiensis BMB171    GCA_000092165.1   \n",
       "69  70                       Bacillus thuringiensis Bt407    GCA_000161495.1   \n",
       "70  71                      Bacillus thuringiensis HD-771    GCA_000292455.1   \n",
       "71  72                      Bacillus thuringiensis HD-789    GCA_000292705.1   \n",
       "72  73                     Bacillus thuringiensis IBL 200    GCA_000161715.1   \n",
       "73  74                        Bacillus thuringiensis MC28    GCA_000300475.1   \n",
       "74  75  Bacillus thuringiensis serovar andalousiensis ...    GCA_000161635.1   \n",
       "75  76  Bacillus thuringiensis serovar berliner ATCC 1...    GCA_000161615.1   \n",
       "76  77     Bacillus thuringiensis serovar chinensis CT-43    GCA_000193355.1   \n",
       "77  78   Bacillus thuringiensis serovar finitimus YBT-020    GCA_000190515.1   \n",
       "78  79  Bacillus thuringiensis serovar huazhongensis B...    GCA_000161675.1   \n",
       "79  80  Bacillus thuringiensis serovar kurstaki str. HD73    GCA_000338755.1   \n",
       "80  81  Bacillus thuringiensis serovar kurstaki str. T...    GCA_000161575.1   \n",
       "81  82  Bacillus thuringiensis serovar monterrey BGSC ...    GCA_000161595.1   \n",
       "82  83  Bacillus thuringiensis serovar pakistani str. ...    GCA_000161555.1   \n",
       "83  84  Bacillus thuringiensis serovar pondicheriensis...    GCA_000161655.1   \n",
       "84  85  Bacillus thuringiensis serovar pulsiensis BGSC...    GCA_000161695.1   \n",
       "85  86  Bacillus thuringiensis serovar thuringiensis s...    GCA_000341665.1   \n",
       "86  87  Bacillus thuringiensis serovar thuringiensis s...    GCA_000161515.1   \n",
       "87  88  Bacillus thuringiensis serovar tochigiensis BG...    GCA_000161475.1   \n",
       "88  89               Bacillus thuringiensis str. Al Hakam    GCA_000015065.1   \n",
       "89  90                         Bacillus velezensis AS43.3    GCA_000319475.1   \n",
       "90  91                       Bacillus velezensis CAU B946    GCA_000283695.1   \n",
       "91  92                          Bacillus velezensis FZB42    GCA_000015785.1   \n",
       "92  93                       Bacillus velezensis UCMB5036    GCA_000341875.1   \n",
       "93  94                   Bacillus velezensis YAU B9601-Y2    GCA_000284395.1   \n",
       "94  95                               Bacillus xiamenensis    GCA_001857925.1   \n",
       "\n",
       "    Gbrs_paired_asm  \n",
       "0   GCF_000196735.1  \n",
       "1   GCF_000242855.2  \n",
       "2   GCF_000204275.1  \n",
       "3   GCF_000195515.1  \n",
       "4   GCF_000221645.1  \n",
       "5   GCF_000022865.1  \n",
       "6   GCF_000007845.1  \n",
       "7   GCF_000021445.1  \n",
       "8   GCF_000258885.1  \n",
       "9   GCF_000008165.1  \n",
       "10  GCF_000742895.1  \n",
       "11  GCF_000022505.1  \n",
       "12  GCF_000832865.1  \n",
       "13  GCF_000160935.1  \n",
       "14  GCF_000161135.1  \n",
       "15  GCF_000161375.1  \n",
       "16  GCF_000161395.1  \n",
       "17  GCF_000003955.1  \n",
       "18  GCF_000021225.1  \n",
       "19  GCF_000161335.1  \n",
       "20  GCF_000160975.1  \n",
       "21  GCF_000161355.1  \n",
       "22  GCF_000021785.1  \n",
       "23  GCF_000160895.1  \n",
       "24  GCF_000008005.1  \n",
       "25  GCF_000007825.1  \n",
       "26  GCF_000161015.1  \n",
       "27  GCF_000021205.1  \n",
       "28  GCF_000161115.1  \n",
       "29  GCF_000161095.1  \n",
       "..              ...  \n",
       "65  GCF_000009045.1  \n",
       "66  GCF_000321395.1  \n",
       "67  GCF_000227485.1  \n",
       "68  GCF_000092165.1  \n",
       "69  GCF_000161495.1  \n",
       "70  GCF_000292455.1  \n",
       "71  GCF_000292705.1  \n",
       "72  GCF_000161715.1  \n",
       "73  GCF_000300475.1  \n",
       "74  GCF_000161635.1  \n",
       "75  GCF_000161615.1  \n",
       "76  GCF_000193355.1  \n",
       "77  GCF_000190515.1  \n",
       "78  GCF_000161675.1  \n",
       "79  GCF_000338755.1  \n",
       "80  GCF_000161575.1  \n",
       "81  GCF_000161595.1  \n",
       "82  GCF_000161555.1  \n",
       "83  GCF_000161655.1  \n",
       "84  GCF_000161695.1  \n",
       "85  GCF_000341665.1  \n",
       "86  GCF_000161515.1  \n",
       "87  GCF_000161475.1  \n",
       "88  GCF_000015065.1  \n",
       "89  GCF_000319475.1  \n",
       "90  GCF_000283695.1  \n",
       "91  GCF_000015785.1  \n",
       "92  GCF_000341875.1  \n",
       "93  GCF_000284395.1  \n",
       "94  GCF_001857925.1  \n",
       "\n",
       "[95 rows x 4 columns]"
      ]
     },
     "execution_count": 87,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "strains"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [],
   "source": [
    "my_strains = literal_eval(students[students[\"Student\"] == \"Vasily Minkov\"].iloc[:, -1].values[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GCA_000204275.1\n",
      "GCA_000221645.1\n",
      "GCA_000242855.2\n",
      "GCA_000008165.1\n",
      "GCA_000021445.1\n",
      "GCA_000022865.1\n",
      "GCA_000832865.1\n",
      "GCA_000161395.1\n",
      "GCA_000160995.1\n",
      "GCA_000161355.1\n",
      "GCA_000161255.1\n",
      "GCA_000003645.1\n",
      "GCA_000160975.1\n",
      "GCA_000003925.1\n",
      "GCA_000017885.4\n",
      "GCA_000209795.2\n",
      "GCA_000186745.1\n",
      "GCA_000161475.1\n",
      "GCA_000161575.1\n",
      "GCA_000193355.1\n",
      "GCA_000190515.1\n",
      "GCA_000161655.1\n",
      "GCA_001857925.1\n",
      "GCA_000283695.1\n"
     ]
    }
   ],
   "source": [
    "for strain in my_strains:\n",
    "    print(strains[\"Assembly accession\"][strain-1])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "metadata": {},
   "outputs": [],
   "source": [
    "path1 = \"/Users/wassilyminkow/Downloads/pan-genome/genomes/\"\n",
    "path2 = \"/Users/wassilyminkow/Downloads/pan-genome/my_genoms/\"\n",
    "\n",
    "call(\"mkdir -p {}\".format(path2), shell=True)\n",
    "for genom in os.listdir(path1):\n",
    "    for strain in my_strains:\n",
    "        if genom.split(\"_\")[1] == strains[\"Assembly accession\"][strain-1].split(\"_\")[1]:\n",
    "            call('mv \"{p1}{g}\" \"{p2}{g}\"'.format(p1=path1, p2=path2, g=genom), shell=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "*****************************************************************\n",
      "\u001b[1;32mProteinortho\u001b[0m with PoFF version 6.0.1 - An orthology detection tool\n",
      "*****************************************************************\n",
      "Detected 4 available CPU threads, Detected 'diamond' version 0.9.24\n",
      "Checking input files.\n",
      "Checking /Users/wassilyminkow/Downloads/pan-genome/my_genoms/GCA_000003645.1_ASM364v1_protein.faa... ok\n",
      "Checking /Users/wassilyminkow/Downloads/pan-genome/my_genoms/GCA_000003925.1_ASM392v1_protein.faa... ok\n",
      "Checking /Users/wassilyminkow/Downloads/pan-genome/my_genoms/GCA_000008165.1_ASM816v1_protein.faa... ok\n",
      "Checking /Users/wassilyminkow/Downloads/pan-genome/my_genoms/GCA_000017885.4_ASM1788v4_protein.faa... ok\n",
      "Checking /Users/wassilyminkow/Downloads/pan-genome/my_genoms/GCA_000021445.1_ASM2144v1_protein.faa... ok\n",
      "Checking /Users/wassilyminkow/Downloads/pan-genome/my_genoms/GCA_000022865.1_ASM2286v1_protein.faa... ok\n",
      "Checking /Users/wassilyminkow/Downloads/pan-genome/my_genoms/GCA_000160975.1_ASM16097v1_protein.faa... ok\n",
      "Checking /Users/wassilyminkow/Downloads/pan-genome/my_genoms/GCA_000160995.1_ASM16099v1_protein.faa... ok\n",
      "Checking /Users/wassilyminkow/Downloads/pan-genome/my_genoms/GCA_000161255.1_ASM16125v1_protein.faa... ok\n",
      "Checking /Users/wassilyminkow/Downloads/pan-genome/my_genoms/GCA_000161355.1_ASM16135v1_protein.faa... ok\n",
      "Checking /Users/wassilyminkow/Downloads/pan-genome/my_genoms/GCA_000161395.1_ASM16139v1_protein.faa... ok\n",
      "Checking /Users/wassilyminkow/Downloads/pan-genome/my_genoms/GCA_000161475.1_ASM16147v1_protein.faa... ok\n",
      "Checking /Users/wassilyminkow/Downloads/pan-genome/my_genoms/GCA_000161575.1_ASM16157v1_protein.faa... ok\n",
      "Checking /Users/wassilyminkow/Downloads/pan-genome/my_genoms/GCA_000161655.1_ASM16165v1_protein.faa... ok\n",
      "Checking /Users/wassilyminkow/Downloads/pan-genome/my_genoms/GCA_000186745.1_ASM18674v1_protein.faa... ok\n",
      "Checking /Users/wassilyminkow/Downloads/pan-genome/my_genoms/GCA_000190515.1_ASM19051v1_protein.faa... ok\n",
      "Checking /Users/wassilyminkow/Downloads/pan-genome/my_genoms/GCA_000193355.1_ASM19335v1_protein.faa... ok\n",
      "Checking /Users/wassilyminkow/Downloads/pan-genome/my_genoms/GCA_000204275.1_ASM20427v1_protein.faa... ok\n",
      "Checking /Users/wassilyminkow/Downloads/pan-genome/my_genoms/GCA_000209795.2_ASM20979v2_protein.faa... ok\n",
      "Checking /Users/wassilyminkow/Downloads/pan-genome/my_genoms/GCA_000221645.1_ASM22164v1_protein.faa... ok\n",
      "Checking /Users/wassilyminkow/Downloads/pan-genome/my_genoms/GCA_000242855.2_ASM24285v2_protein.faa... ok\n",
      "Checking /Users/wassilyminkow/Downloads/pan-genome/my_genoms/GCA_000283695.1_ASM28369v1_protein.faa... ok\n",
      "Checking /Users/wassilyminkow/Downloads/pan-genome/my_genoms/GCA_000832865.1_ASM83286v1_protein.faa... ok\n",
      "Checking /Users/wassilyminkow/Downloads/pan-genome/my_genoms/GCA_001857925.1_ASM185792v1_protein.faa... ok\n",
      "\n",
      "\u001b[1;32m**Step 1**\u001b[0m\n",
      "Generating indices.\n",
      "Building database for '/Users/wassilyminkow/Downloads/pan-genome/my_genoms/GCA_000193355.1_ASM19335v1_protein.faa'\t(6206 sequences)\n",
      "Building database for '/Users/wassilyminkow/Downloads/pan-genome/my_genoms/GCA_000832865.1_ASM83286v1_protein.faa'\t(6130 sequences)\n",
      "Building database for '/Users/wassilyminkow/Downloads/pan-genome/my_genoms/GCA_000161655.1_ASM16165v1_protein.faa'\t(6053 sequences)\n",
      "Building database for '/Users/wassilyminkow/Downloads/pan-genome/my_genoms/GCA_000021445.1_ASM2144v1_protein.faa'\t(5896 sequences)\n",
      "Building database for '/Users/wassilyminkow/Downloads/pan-genome/my_genoms/GCA_000161395.1_ASM16139v1_protein.faa'\t(5810 sequences)\n",
      "Building database for '/Users/wassilyminkow/Downloads/pan-genome/my_genoms/GCA_000190515.1_ASM19051v1_protein.faa'\t(5782 sequences)\n",
      "Building database for '/Users/wassilyminkow/Downloads/pan-genome/my_genoms/GCA_000161475.1_ASM16147v1_protein.faa'\t(5732 sequences)\n",
      "Building database for '/Users/wassilyminkow/Downloads/pan-genome/my_genoms/GCA_000160975.1_ASM16097v1_protein.faa'\t(5667 sequences)\n",
      "Building database for '/Users/wassilyminkow/Downloads/pan-genome/my_genoms/GCA_000003925.1_ASM392v1_protein.faa'\t(5658 sequences)\n",
      "Building database for '/Users/wassilyminkow/Downloads/pan-genome/my_genoms/GCA_000161355.1_ASM16135v1_protein.faa'\t(5612 sequences)\n",
      "Building database for '/Users/wassilyminkow/Downloads/pan-genome/my_genoms/GCA_000160995.1_ASM16099v1_protein.faa'\t(5587 sequences)\n",
      "Building database for '/Users/wassilyminkow/Downloads/pan-genome/my_genoms/GCA_000161575.1_ASM16157v1_protein.faa'\t(5556 sequences)\n",
      "Building database for '/Users/wassilyminkow/Downloads/pan-genome/my_genoms/GCA_000022865.1_ASM2286v1_protein.faa'\t(5291 sequences)\n",
      "Building database for '/Users/wassilyminkow/Downloads/pan-genome/my_genoms/GCA_000008165.1_ASM816v1_protein.faa'\t(5287 sequences)\n",
      "Building database for '/Users/wassilyminkow/Downloads/pan-genome/my_genoms/GCA_000003645.1_ASM364v1_protein.faa'\t(5282 sequences)\n",
      "Building database for '/Users/wassilyminkow/Downloads/pan-genome/my_genoms/GCA_000161255.1_ASM16125v1_protein.faa'\t(4750 sequences)\n",
      "Building database for '/Users/wassilyminkow/Downloads/pan-genome/my_genoms/GCA_000209795.2_ASM20979v2_protein.faa'\t(4477 sequences)\n",
      "Building database for '/Users/wassilyminkow/Downloads/pan-genome/my_genoms/GCA_000204275.1_ASM20427v1_protein.faa'\t(4228 sequences)\n",
      "Building database for '/Users/wassilyminkow/Downloads/pan-genome/my_genoms/GCA_000221645.1_ASM22164v1_protein.faa'\t(4190 sequences)\n",
      "Building database for '/Users/wassilyminkow/Downloads/pan-genome/my_genoms/GCA_000186745.1_ASM18674v1_protein.faa'\t(4145 sequences)\n",
      "Building database for '/Users/wassilyminkow/Downloads/pan-genome/my_genoms/GCA_000242855.2_ASM24285v2_protein.faa'\t(3860 sequences)\n",
      "Building database for '/Users/wassilyminkow/Downloads/pan-genome/my_genoms/GCA_000283695.1_ASM28369v1_protein.faa'\t(3823 sequences)\n",
      "Building database for '/Users/wassilyminkow/Downloads/pan-genome/my_genoms/GCA_000017885.4_ASM1788v4_protein.faa'\t(3710 sequences)\n",
      "Building database for '/Users/wassilyminkow/Downloads/pan-genome/my_genoms/GCA_001857925.1_ASM185792v1_protein.faa'\t(3556 sequences)\n",
      "\n",
      "\u001b[1;32m**Step 2**\u001b[0m\n",
      "Running blast analysis: 100% (276/276)                                         \n",
      "[OUTPUT] -> written to test.blast-graph\n",
      "\n",
      "\u001b[1;32m**Step 3**\u001b[0m\n",
      "Clustering by similarity (Proteinortho mode) using 16384 MB of memory (default value, command 'free' not found) and 4 cpu core(s).\n",
      "Reading test.blast-graph\n",
      "24 species\n",
      "114291 paired proteins\n",
      "896322 bidirectional edges\n",
      "Done                       \n",
      "[OUTPUT] -> Orthologous groups are written to test.proteinortho.tsv\n",
      " You can extract the fasta files of each orthology group with 'proteinortho_grab_proteins.pl -tofiles test.proteinortho.tsv /Users/wassilyminkow/Downloads/pan-genome/my_genoms/GCA_000193355.1_ASM19335v1_protein.faa /Users/wassilyminkow/Downloads/pan-genome/my_genoms/GCA_000832865.1_ASM83286v1_protein.faa /Users/wassilyminkow/Downloads/pan-genome/my_genoms/GCA_000161655.1_ASM16165v1_protein.faa /Users/wassilyminkow/Downloads/pan-genome/my_genoms/GCA_000021445.1_ASM2144v1_protein.faa /Users/wassilyminkow/Downloads/pan-genome/my_genoms/GCA_000161395.1_ASM16139v1_protein.faa /Users/wassilyminkow/Downloads/pan-genome/my_genoms/GCA_000190515.1_ASM19051v1_protein.faa /Users/wassilyminkow/Downloads/pan-genome/my_genoms/GCA_000161475.1_ASM16147v1_protein.faa /Users/wassilyminkow/Downloads/pan-genome/my_genoms/GCA_000160975.1_ASM16097v1_protein.faa /Users/wassilyminkow/Downloads/pan-genome/my_genoms/GCA_000003925.1_ASM392v1_protein.faa /Users/wassilyminkow/Downloads/pan-genome/my_genoms/GCA_000161355.1_ASM16135v1_protein.faa /Users/wassilyminkow/Downloads/pan-genome/my_genoms/GCA_000160995.1_ASM16099v1_protein.faa /Users/wassilyminkow/Downloads/pan-genome/my_genoms/GCA_000161575.1_ASM16157v1_protein.faa /Users/wassilyminkow/Downloads/pan-genome/my_genoms/GCA_000022865.1_ASM2286v1_protein.faa /Users/wassilyminkow/Downloads/pan-genome/my_genoms/GCA_000008165.1_ASM816v1_protein.faa /Users/wassilyminkow/Downloads/pan-genome/my_genoms/GCA_000003645.1_ASM364v1_protein.faa /Users/wassilyminkow/Downloads/pan-genome/my_genoms/GCA_000161255.1_ASM16125v1_protein.faa /Users/wassilyminkow/Downloads/pan-genome/my_genoms/GCA_000209795.2_ASM20979v2_protein.faa /Users/wassilyminkow/Downloads/pan-genome/my_genoms/GCA_000204275.1_ASM20427v1_protein.faa /Users/wassilyminkow/Downloads/pan-genome/my_genoms/GCA_000221645.1_ASM22164v1_protein.faa /Users/wassilyminkow/Downloads/pan-genome/my_genoms/GCA_000186745.1_ASM18674v1_protein.faa /Users/wassilyminkow/Downloads/pan-genome/my_genoms/GCA_000242855.2_ASM24285v2_protein.faa /Users/wassilyminkow/Downloads/pan-genome/my_genoms/GCA_000283695.1_ASM28369v1_protein.faa /Users/wassilyminkow/Downloads/pan-genome/my_genoms/GCA_000017885.4_ASM1788v4_protein.faa /Users/wassilyminkow/Downloads/pan-genome/my_genoms/GCA_001857925.1_ASM185792v1_protein.faa'\n",
      " (Careful: This will generate a file foreach line in the file test.proteinortho.tsv).\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[OUTPUT] -> Orthologous pairs are written to test.proteinortho-graph\n",
      "[OUTPUT] -> Orthologous groups are written to test.proteinortho.html\n",
      "\n",
      "\u001b[1;32mAll finished.\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "! proteinortho -project=test /Users/wassilyminkow/Downloads/pan-genome/my_genoms/*faa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
